### DISCUSSION TITLE
'darcy_thermo_mech' tutorials not compiling, but 'examples' are compiling?

### DISCUSSION THREAD

**Discussion by kurtenkera** [idaholab/moose#26832] [Q&A Getting Started]
I'm working on a Linux laptop and following the MOOSE VIrtual Workshop (Summer 2020), and for some reason I cannot compile the darcy_thermo_mech/step01_diffusion file. 

I have made sure that I am in the correct directory, which is /projects/moose/tutorials/darcy_thermo_mech/step01_diffusion. I have also made sure that I have executed `conda activate moose`. However, when I execute the command `make`, I get the following error message/output:

```
Not compiling MOOSE with NEML2 because NEML2_DIR is not a valid NEML2 checkout and/or libtorch is not enabled.
Linking Library /home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//modules/module_loader/lib/libmodule_loader_with_ray_ht_tm_pf-opt.la...
libtool: warning: library '/home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//modules/ray_tracing/lib/libray_tracing-opt.la' was moved.
libtool: warning: library '/home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//modules/heat_transfer/lib/libheat_transfer-opt.la' was moved.
/bin/grep: /home/enk001/projects_MOOSE/moose/tutorials/darcy_thermo_mech/../..//modules/ray_tracing/lib/libray_tracing-opt.la: No such file or directory
/bin/sed: can't read /home/enk001/projects_MOOSE/moose/tutorials/darcy_thermo_mech/../..//modules/ray_tracing/lib/libray_tracing-opt.la: No such file or directory
libtool:   error: '/home/enk001/projects_MOOSE/moose/tutorials/darcy_thermo_mech/../..//modules/ray_tracing/lib/libray_tracing-opt.la' is not a valid libtool archive
make: *** [/home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//framework/app.mk:370: /home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//modules/module_loader/lib/libmodule_loader_with_ray_ht_tm_pf-opt.la] Error 1
```

It is important to mention that after I installed MOOSE, I changed the 'projects' directory to 'projects_MOOSE' just as I thought this was a better name. But then as I realised a lot of the examples on the MOOSE website run with the directory titled 'projects', I changed the name back. But even still, I was able to compile the first three examples in projects/moose/examples after doing this and they all worked perfectly fine? So how can I get the tutorial examples to work/compile?

Note: I changed the 'projects' directory back to 'projects_MOOSE' and then tried to compile the step01_diffusion application again, and got a slightly different error message (see below). It seems like MOOSE doesn't like the fact that I've changed the 'projects' directory to 'projects_MOOSE' after installation?

```
Not compiling MOOSE with NEML2 because NEML2_DIR is not a valid NEML2 checkout and/or libtorch is not enabled.
Symlinking MOOSE configure /home/enk001/projects_MOOSE/moose/tutorials/darcy_thermo_mech/../..//framework/build/header_symlinks/MooseConfig.h
Linking Library /home/enk001/projects_MOOSE/moose/tutorials/darcy_thermo_mech/../..//modules/module_loader/lib/libmodule_loader_with_ray_ht_tm_pf-opt.la...
libtool: warning: library '/home/enk001/projects_MOOSE/moose/tutorials/darcy_thermo_mech/../..//modules/tensor_mechanics/lib/libtensor_mechanics-opt.la' was moved.
/bin/grep: /home/enk001/projects/moose/modules/tensor_mechanics/lib/libtensor_mechanics-opt.la: No such file or directory
/bin/sed: can't read /home/enk001/projects/moose/modules/tensor_mechanics/lib/libtensor_mechanics-opt.la: No such file or directory
libtool:   error: '/home/enk001/projects/moose/modules/tensor_mechanics/lib/libtensor_mechanics-opt.la' is not a valid libtool archive
make: *** [/home/enk001/projects_MOOSE/moose/tutorials/darcy_thermo_mech/../..//framework/app.mk:370: /home/enk001/projects_MOOSE/moose/tutorials/darcy_thermo_mech/../..//modules/module_loader/lib/libmodule_loader_with_ray_ht_tm_pf-opt.la] Error 1
```

I was also curious: Do Peacock and Paraview come with the MOOSE installation? Or do I have to install them separately to view results generated in MOOSE? 

Thanks in advance!

**Comment by GiudGiud** [comment]
Hello

When did you download moose?
The tutorials were not building for a week last week, that got fixed

Guillaume

**Reply by kurtenkera** [reply]
I downloaded MOOSE on Friday (two days ago)? Should I update conda and MOOSE and retry? Am I right in saying that it shouldn't matter that I changed the directory name from 'projects' to 'projects_MOOSE'?

If I update conda and MOOSE, do you know how much GB this will use? I am currently hotspotting my laptop's internet connection with my mobile phone, and I have a monthly limit of 100 GB so I don't want to chew into this too much?

**Reply by GiudGiud** [reply]
Friday should have been fine. 

I'll check the tutorials when I get back.

For now can you use the next branch instead of the master branch (use git checkout next)

**Reply by GiudGiud** [reply]
Renaming the projects folder is totally ok

And a full moose and libmesh will be a couple GB if you started from fresh. For an update probably under a GB

**Reply by GiudGiud** [reply]
Renaming the projects folder is totally ok

And a full moose and libmesh will be a couple GB if you started from fresh. For an update probably under a GB

**Reply by kurtenkera** [reply]
I switched to the next branch using `git checkout next`, but ultimately I got a similar error message (see below) when trying to compile step01_diffusion:

```
Not compiling MOOSE with NEML2 because NEML2_DIR is not a valid NEML2 checkout and/or libtorch is not enabled.
Checking if header needs updating: /home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//framework/include/base/MooseRevision.h...
Symlinking MOOSE configure /home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//framework/build/header_symlinks/MooseConfig.h
Creating Unity /home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//framework/build/unity_src/userobjects_Unity.C
Compiling C++ (in opt mode) /home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//framework/build/unity_src/userobjects_Unity.C...
Linking Library /home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//framework/libmoose-opt.la...
Linking Library /home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//modules/module_loader/lib/libmodule_loader_with_ray_ht_tm_pf-opt.la...
libtool: warning: library '/home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//modules/ray_tracing/lib/libray_tracing-opt.la' was moved.
libtool: warning: library '/home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//modules/heat_transfer/lib/libheat_transfer-opt.la' was moved.
/bin/grep: /home/enk001/projects_MOOSE/moose/tutorials/darcy_thermo_mech/../..//modules/ray_tracing/lib/libray_tracing-opt.la: No such file or directory
/bin/sed: can't read /home/enk001/projects_MOOSE/moose/tutorials/darcy_thermo_mech/../..//modules/ray_tracing/lib/libray_tracing-opt.la: No such file or directory
libtool:   error: '/home/enk001/projects_MOOSE/moose/tutorials/darcy_thermo_mech/../..//modules/ray_tracing/lib/libray_tracing-opt.la' is not a valid libtool archive
make: *** [/home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//framework/app.mk:370: /home/enk001/projects/moose/tutorials/darcy_thermo_mech/../..//modules/module_loader/lib/libmodule_loader_with_ray_ht_tm_pf-opt.la] Error 1
```

**Reply by GiudGiud** [reply]
Some of the warnings are detecting this moving around.

Did you build moose then change the project folder name?

If so you want to git clean -xfD the repo then build again (save all your work first with git commit -a)

**Reply by kurtenkera** [reply]
Yes I built MOOSE using the following code:

```
mkdir -p ~/projects
cd ~/projects
git clone https://github.com/idaholab/moose.git
cd moose
git checkout master
```

Then I changed the directory 'projects' to 'projects_MOOSE'.

**Reply by GiudGiud** [reply]
Did you 'make' before or after changing the projects folder?

**Reply by kurtenkera** [reply]
So would I execute the following code? (see below):

```
git clean -xfD
git commit -a
git clone https://github.com/idaholab/moose.git
cd moose
git checkout master
```

I'm kind of new to git so learning this too.

**Reply by kurtenkera** [reply]
I executed the code below immediately after installing MOOSE.
```
cd ~/projects/moose/test
make -j 6
```
I then changed the projects folder and then tried to 'make' with the tutorial step01_diffusion example and it failed. But I was able to 'make' the first three 'examples' even after changing the projects folder?

**Reply by GiudGiud** [reply]
The git commit is the first command to execute
And you do not need to clone the repository again. Just use 'make' again after cleaning


The makefiles are not uniformly resistant to changing the root folder. The rename caused the build failure. 

Either cleaning or undoing the rename will fix the problem

**Reply by kurtenkera** [reply]
Thanks for your help. This seems to have worked; however, I don't understand what git clean -xfD accomplished? Shouldn't this have deleted all files in the projects working directory (including the moose directory)? Yet when I executed this command, the 'moose' directory and its contents still remained? Very confusing...

**Reply by GiudGiud** [reply]
Git clean only cleans git repositories. It should have been run in the moose directory not projects

**Reply by kurtenkera** [reply]
And so would this have deleted the untracked files that were created when I tried to compile the step01_diffusion example (after changing the projects directory to projects_MOOSE)? But the reason all other files weren't deleted is because they were being tracked already?

**Reply by GiudGiud** [reply]
yes
